nameko/containers.py

```diff
@@ -134,8 +144,16 @@ def __init__(self, service_cls, config, worker_ctx_cls=None):
         self.service_cls = service_cls
         self.config = config
 
-        if worker_ctx_cls is None:
-            worker_ctx_cls = WorkerContext
+        if worker_ctx_cls is not None:
+            warnings.warn(
```

Comment:
we are starting to add a whole bunch of deprecation warnings. we might want some kind of plan for how long we keep the old behaviours around, and maybe add a grep-able comment marker to tell ourselves when (which version) these can be tidied up